test/Test.re

```diff
@@ -768,9 +769,122 @@ let core = [
       );
     },
   ),
+  (
+    "Test transition from empty list to non-empty list",
+    `Quick,
+    () => {
+      GlobalState.useTailHack := false;
```

Comment:
@wokalski are you happy to mutate `GlobalState.useTailHack` in tests? Right now, there is no isolation, i.e. the next test will be run with whatever value the previous test left `GlobalState.useTailHack` in, correct?